### PR TITLE
Bump CAPI to v1.4.x

### PR DESCRIPTION
**Pull request by kashifest**

### CODE REVIEW

**Review by lentzi90 — COMMENTED** [review]
/approve

**Comment by kashifest** [comment]
> Needs [metal3-io/cluster-api-provider-metal3#888](https://github.com/metal3-io/cluster-api-provider-metal3/pull/888) to go in first

We might need to take this PR first but via force merging.

File: lib/common.sh [hunk 1]
```diff
@@ -136,7 +136,7 @@ export BMO_BASE_URL="${BMO_BASE_URL:-metal3-io/baremetal-operator}"
 export RUN_LOCAL_IRONIC_SCRIPT="${BMOPATH}/tools/run_local_ironic.sh"
 
 export CAPM3PATH="${CAPM3PATH:-${M3PATH}/cluster-api-provider-metal3}"
-export CAPM3_BASE_URL="${CAPM3_BASE_URL:-metal3-io/cluster-api-provider-metal3}"
+export CAPM3_BASE_URL="Nordix/cluster-api-provider-metal3"
```
Comment (tuminoid) [inline lib/common.sh:139]: Tip: It is much cleaner to put the wanted test configuration in `config_example.sh` than change them all over the place in `lib`.

File: lib/releases.sh [hunk 2]
```diff
@@ -58,9 +58,14 @@ elif [ "${CAPM3RELEASEBRANCH}" == "release-1.1" ]; then
 elif [ "${CAPM3RELEASEBRANCH}" == "release-1.2" ]; then
   export CAPM3RELEASE="${CAPM3RELEASE:-$(get_latest_release "${CAPM3RELEASEPATH}" "v1.2.")}"
   export CAPIRELEASE="${CAPIRELEASE:-$(get_latest_release "${CAPIRELEASEPATH}" "v1.2.")}"
-else
+elif [ "${CAPM3RELEASEBRANCH}" == "release-1.3" ]; then
```
Comment (tuminoid) [inline lib/releases.sh:61]: Let's not add any  new `[` and `==` combinations even if the previous ones have it.

File: lib/common.sh (hunk 1 above)
Comment (kashifest) [reply lib/common.sh:139]: its a test commit. Thanks for the tip .

**Review by furkatgofurov7 — DISMISSED** [review]
Looks good, please remove the test commit once it serves its purpose

/approve

**Comment by kashifest** [comment]
Passed tests: 
https://jenkins.nordix.org/job/metal3_metal3_dev_env_main_integration_test_centos/50/
https://jenkins.nordix.org/job/metal3_metal3_dev_env_release-1-3_integration_test_centos/162/
https://jenkins.nordix.org/job/metal3_metal3_dev_env_main_e2e_upgrade_test_ubuntu/1/
https://jenkins.nordix.org/job/metal3_metal3_dev_env_main_e2e_feature_test_ubuntu/2/
https://jenkins.nordix.org/job/metal3_metal3_dev_env_e2e_integration_test_ubuntu/99/
https://jenkins.nordix.org/job/metal3_metal3_dev_env_main_integration_test_ubuntu/269/

File: lib/releases.sh (hunk 2 above)
Comment (kashifest) [reply lib/releases.sh]: is the change ok now @tuminoid ?

**Review by furkatgofurov7 — COMMENTED** [review]
/lgtm

**Comment by kashifest** [comment]
Tests have passed here with test commit. However, tests wont pass here until https://github.com/metal3-io/cluster-api-provider-metal3/pull/888 and perhaps https://github.com/metal3-io/ip-address-manager/pull/214 lands. So we have to force merge this PR .

**Pull request merged**